crates/precompile/src/xcalloptions.rs

```diff
@@ -29,14 +30,31 @@ fn xcalloptions_run(input: &[u8], _gas_limit: u64, _env: &Env, call_options: &mu
     }
 
     // Set the call options
-    *call_options = Some(CallOptions {
-        chain_id,
-        sandbox,
-        tx_origin: ChainAddress(chain_id, tx_origin),
-        msg_sender: ChainAddress(chain_id, msg_sender),
-        block_hash,
-        proof: proof.to_vec(),
-    });
-
-    Ok(PrecompileOutput::new(0, Bytes::default()))
+    // *call_options = Some(CallOptions {
+    //     chain_id,
+    //     sandbox,
+    //     tx_origin: ChainAddress(chain_id, tx_origin),
+    //     msg_sender: ChainAddress(chain_id, msg_sender),
+    //     block_hash,
+    //     proof: proof.to_vec(),
+    // });
+    // println!("CallOptions: {:?}", call_options);
+
+    let mut prefix = b"XCallOptions".to_vec();
+    prefix.extend_from_slice(&input);
+
+    Ok(PrecompileOutput::new(0, Bytes::copy_from_slice(&prefix)))
```

Comment:
The value can be easily fixed in something like the identity precompile where if you would send this input it gets copied to the output and then revm would also pick it up as call options, which would be bad.

The output also impacts the actual return value inside the EVM, so not ideal because can increase gas costs (though maybe not if it's not read, not sure with precompiles), ideally this data is not exposed to the actual EVM.

Reply:
i know i know
just a temp solution, I'm gonna directly change `PrecompileOutput` into a enum or something